iOS-Weather-App/AppDelegate.swift

```diff
@@ -10,27 +10,45 @@ import UIKit
 @main
 class AppDelegate: UIResponder, UIApplicationDelegate {
 
-
+    var window: UIWindow?
+    private var navigationController: UINavigationController?
+    private let weatherController = WeatherViewController()
+    private let searchViewModel = SearchViewModel()
+    private let currentDetailedWeatherModel = CurrentDetailedWeatherViewModel()
+    private let searchController = SearchViewController()
+    private let currentDetailsController = CurrentDetailedWeatherViewController()
 
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
-        // Override point for customization after application launch.
-        return true
-    }
 
-    // MARK: UISceneSession Lifecycle
 
-    func application(_ application: UIApplication, configurationForConnecting connectingSceneSession: UISceneSession, options: UIScene.ConnectionOptions) -> UISceneConfiguration {
-        // Called when a new scene session is being created.
-        // Use this method to select a configuration to create the new scene with.
-        return UISceneConfiguration(name: "Default Configuration", sessionRole: connectingSceneSession.role)
-    }
+        window = UIWindow(frame: UIScreen.main.bounds)
+        window?.makeKeyAndVisible()
 
-    func application(_ application: UIApplication, didDiscardSceneSessions sceneSessions: Set<UISceneSession>) {
-        // Called when the user discards a scene session.
-        // If any sessions were discarded while the application was not running, this will be called shortly after application:didFinishLaunchingWithOptions.
-        // Use this method to release any resources that were specific to the discarded scenes, as they will not return.
-    }
+        weatherController.configure(searchViewModel: searchViewModel, currentDetailedWeatherModel: currentDetailedWeatherModel)
```

Comment:
weatherController should contain only weatherViewModel, shouldn't pass all viewModels